src/module.c

```diff
@@ -858,9 +858,24 @@ int TSDB_incrby(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     if (RMUtil_ParseArgs(argv, argc, 2, "d", &incrby) != REDISMODULE_OK)
         return RedisModule_WrongArity(ctx);
 
+
+    long long currentUpdatedTime = -1;
+    int timestampLoc = RMUtil_ArgIndex("TIMESTAMP", argv, argc);
+    if (timestampLoc > 0) {
+        if ((RedisModule_StringToLongLong(argv[timestampLoc + 1], (long long *) &currentUpdatedTime) != REDISMODULE_OK)) {
```

Comment:
@ashtul I think going positive it better
```C
if (timestampLoc > 0) {
        if ((RedisModule_StringToLongLong(argv[timestampLoc + 1], (long long *) &currentUpdatedTime) == REDISMODULE_OK)) {
            currentUpdatedTime = RedisModule_Milliseconds();
        } else if (RMUtil_StringEqualsC(argv[timestampLoc + 1], "*")) { 
            // if timestamp is "*", take current time (automatic timestamp)
            currentUpdatedTime = RedisModule_Milliseconds();
        } else {
            return RedisModule_ReplyWithError(ctx, "TSDB: invalid timestamp");
        }
} else {
        currentUpdatedTime = RedisModule_Milliseconds();
}
```

Reply:
@gkorland 
He never uses it this way in the code. I took as a code convention.
It is the same complexity and I agree your code is more elegant. It is your call.

Reply:
so please refactor it

Reply:
Done